### PR TITLE
Woa 3 leaderboard

### PR DESCRIPTION
**Pull request by sandeepsuri**
## PR Submission - Header and Leaderboard Implementation

**Hello team,**

I'm submitting this pull request for review to share the progress I've made on the header and leaderboard components of our app. Here are the key highlights of this PR:

**Header Implementation:**
- Added a header for the app to provide a clear and consistent visual identity.

**Leaderboard Implementation:**
- Created a leaderboard with hardcoded information to showcase the initial design.
- Implemented a toggle feature for collapsing and expanding the leaderboard.
- When expanded, the leaderboard displays everyone in the competition along with their rankings.
- Please note that this is frontend work; no backend connection has been established yet.

I would appreciate your feedback on the design and functionality of these components. Specifically, I'd like to know if the visual elements align with our UI/UX guidelines and if the user interactions are intuitive.

I understand that the next steps involve integrating this with the backend and making the leaderboard dynamically load data. However, for now, I'd like to focus on the front-end and design aspects.

Thank you for your time and input. Please review the code and provide your comments and suggestions.

This is the initial high fidelity design created on figma:
Screenshot 2023-10-24 at 4 57 16 PM

This is the final design I was able to achieve on the branch:
Screenshot 2023-10-24 at 5 00 01 PM

### CODE REVIEW

**Review by sidhuharvir — CHANGES_REQUESTED** [review]
Hello team,

I've reviewed the pull request and I appreciate the work done on the header and leaderboard components. Here's my feedback:

1. **Header Implementation:**
   - The header is a good addition and aligns with our app's visual identity.
   
2. **Leaderboard Implementation:**
   - The leaderboard looks great with the initial design.
   - The toggle feature is a nice touch for user interaction.
   - One issue to note: Circle 5 is a different shape than the rest. This inconsistency should be addressed for a uniform design.
   -  Suggestion: drop the last name and keep only the last name initial, I think it's a good idea for a cleaner and more concise design. It would be great to implement this change.

As for the issue of simultaneous changes to the same file, I recommend that we establish a clear process for version control and collaboration to avoid such conflicts in the future. One way to handle this is to create branches for specific features or components and then merge them into the main development branch once they are reviewed and approved. This will help ensure that changes are made in an organized and controlled manner.

Thank you for your work on this, and I look forward to the improvements. Please proceed with the suggested changes and let's work on a better collaboration process to prevent future conflicts.

Best regards.

File: workout-app/src/components/weekpot.tsx [hunk 1]
```diff
@@ -1,11 +1,71 @@
-import React from 'react'
+import React, { useState } from 'react'
+import "@fontsource/reenie-beanie"
+
+const weekpot = ({toggleBoard, handleToggle}) => {
+
+  const nameMapping = {
+    1: 'Sandeep Suri',
```
Comment (jimip85) [inline workout-app/src/components/weekpot.tsx:7]: Great Work Sunny. I'm assuming once we generate the intake form you will introduce logic to make this dynamic right?

File: workout-app/src/components/weekpot.tsx [hunk 2]
```diff
@@ -1,11 +1,71 @@
...
+    6: 'Gurkirat Gill'
+  }
 
-const weekpot = () => {
   return (
     <div>
-        <div className=" bg-red-400">Week/Pot view</div> 
+      { !toggleBoard ? (
+        <div className="w-80 h-40 justify-center items-center inline-flex" onClick={handleToggle}>
+          <div className="w-80 h-40 bg-slate-800 rounded-2xl border-l-4 border-r-4 border-t-8 border-b-8 border-cyan-900 flex-col justify-center items-center inline-flex">
```
Comment (jimip85) [inline workout-app/src/components/weekpot.tsx]: Might have some redundant code here. Look into this.

File: workout-app/src/components/weekpot.tsx [hunk 3]
```diff
@@ -1,11 +1,71 @@
...
+
+const weekpot = ({toggleBoard, handleToggle}) => {
+
+  const nameMapping = {
+    1: 'Sandeep Suri',
+    2: 'Harvir Sidhu',
+    3: 'Jimi Patel',
+    4: 'Deshane Medley',
+    5: 'Shanthosh Yogendren',
+    6: 'Gurkirat Gill'
```
Comment (jimip85) [inline workout-app/src/components/weekpot.tsx]: Great Work Sunny. I'm assuming once we generate the intake form you will introduce logic to make this dynamic right?

**Review by jimip85 — COMMENTED** [review]
Great job, Sunny. We're almost ready to merge, but I have a few minor changes in mind to ensure everything is perfect.

File: workout-app/src/components/weekpot.tsx (hunk 3 above)
Comment (sandeepsuri) [reply workout-app/src/components/weekpot.tsx]: Hey both @jimip85 @sidhuharvir I looked into your requests and applied those changes. Please take a look at the new changes and let me know if it's good to merge!

**Review by jimip85 — COMMENTED** [review]
Excellent work, Sunny! Your changes appear to be in great shape for merging into the main branch.

**Review by jimip85 — APPROVED** [review]

**Review by sidhuharvir — APPROVED** [review]
Everything looks good.

**Pull request merged**